diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/classes/StatelessClassesRule.kt

```diff
@@ -0,0 +1,99 @@
+import com.pinterest.ktlint.core.ast.ElementType.FILE
+import com.pinterest.ktlint.core.ast.ElementType.FUN
+import com.pinterest.ktlint.core.ast.ElementType.IDENTIFIER
+import com.pinterest.ktlint.core.ast.ElementType.INTERFACE_KEYWORD
+import com.pinterest.ktlint.core.ast.ElementType.OBJECT_DECLARATION
+import com.pinterest.ktlint.core.ast.ElementType.OBJECT_KEYWORD
+import com.pinterest.ktlint.core.ast.ElementType.SUPER_TYPE_ENTRY
+import com.pinterest.ktlint.core.ast.ElementType.SUPER_TYPE_LIST
+import com.pinterest.ktlint.core.ast.children
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.ruleset.constants.EmitType
+import org.cqfn.diktat.ruleset.constants.Warnings
+import org.cqfn.diktat.ruleset.utils.findAllNodesWithSpecificType
+import org.cqfn.diktat.ruleset.utils.getAllChildrenWithType
+import org.cqfn.diktat.ruleset.utils.getAllLeafsWithSpecificType
+import org.cqfn.diktat.ruleset.utils.getFirstChildWithType
+import org.cqfn.diktat.ruleset.utils.hasChildOfType
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.CompositeElement
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.LeafPsiElement
+import org.jetbrains.kotlin.psi.KtClass
+
+/**
+ * This rule checks if class is stateless and if so changes it to object.
+ */
+class StatelessClassesRule(private val configRule: List<RulesConfig>) : Rule("stateless-class") {
+    private lateinit var emitWarn: EmitType
+    private var isFixMode: Boolean = false
+
+    override fun visit(
+            node: ASTNode,
+            autoCorrect: Boolean,
+            emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit
+    ) {
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        val interfaces: MutableList<ASTNode> = mutableListOf()
+
+        // Fixme: We should find interfaces in all project and then check them
+        if (node.elementType == FILE) {
+            val interfacesNodes = node
+                    .findAllNodesWithSpecificType(CLASS)
+                    .filter { it.hasChildOfType(INTERFACE_KEYWORD) }
+            interfacesNodes.forEach { interfaces.add(it) }
+            (node.findAllNodesWithSpecificType(CLASS) subtract interfacesNodes)
+                    .forEach{handleClass(it, interfaces)}
+        }
+    }
+
+    @Suppress("UnsafeCallOnNullableType")
+    private fun handleClass(node: ASTNode, interfaces: MutableList<ASTNode>) {
+        if (isClassExtendsValidInterface(node, interfaces) && isStatelessClass(node)) {
+            Warnings.OBJECT_IS_PREFERRED.warnAndFix(configRule, emitWarn, isFixMode,
+                    "class ${(node.psi as KtClass).name!!}", node.startOffset, node) {
+                val newObjectNode = CompositeElement(OBJECT_DECLARATION)
+                node.treeParent.addChild(newObjectNode, null)
+                node.removeChild(node.firstChildNode)
```

Comment:
`node` is `CLASS` here, right? How does this play with KDocs on the class? Please add such tests.

Reply:
Fixed